src/fleet/azext_fleet/_helpers.py

```diff
@@ -133,10 +133,12 @@ def _handle_merge(existing, addition, key, replace):
 
 def _load_kubernetes_configuration(filename):
     try:
-        with open(filename) as stream:
+        with open(filename, encoding='utf-8') as stream:
             return yaml.safe_load(stream)
     except (IOError, OSError) as ex:
         if getattr(ex, 'errno', 0) == errno.ENOENT:
-            raise CLIError(f'{filename} does not exist')
+            raise CLIError(f'{filename} does not exist') from ex
     except (yaml.parser.ParserError, UnicodeDecodeError) as ex:
-        raise CLIError(f'Error parsing {filename} ({str(ex)})')
+        raise CLIError(f'Error parsing {filename} ({str(ex)})') from ex
+
+    return None
```

Comment:
instead of returning None, does it work instead to `raise` in the (IOError, OSError) case if the error isn't ENOENT?